# Telomerecat (Telomere Computational Analysis Tool)

Telomerecat is a tool for estimating the average telomere length (TL) for a paired end, whole genome sequencing (WGS) sample.

Telomerecat is adaptable, accurate and fast. The algorithm accounts for sequencing amplification artifacts, anneouploidy (common in cancer samples) and noise generated by WGS. For a high coverage WGS BAM file of around 100GB telomerecat can produce an estimate in ~1 hour.

A full manual for Telomerecat is available [here](telomerecat.readthedocs.org).

## INSTALL

Installation is via `pip`.  Simply execute with the path to the packaged distribution:

```bash
pip install --find-links=~/wheels telomerecat
# or
pip install https://github.com/cancerit/telomerecat/archive/master.tar.gz
```

### Package Dependancies

`pip` will install the relevant dependancies, listed here for convenience:

* [parabam](https://github.com/cancerit/parabam)
* [numpy](https://numpy.org/)
* [pysam](https://www.scipy.org/)
* [pandas](https://pandas.pydata.org/)

### Development Dependencies

#### Setup VirtualEnv

```bash
cd $PROJECTROOT
hash virtualenv || pip3 install virtualenv
virtualenv -p python3 env
source env/bin/activate
pip install -r requirements.txt
python setup.py develop # so bin scripts can find module
```

### Cutting a release

The release is handled by wheel:

```bash
$ source env/bin/activate # if not already
$ python setup.py bdist_wheel -d dist
# this creates an wheel archive which can be copied to a deployment location, e.g.
$ scp telomerecat-3.2-py3-none-any.whl user@host:~/wheels
# on host
$ pip install --find-links=~/wheels telomerecat
```
